### PR TITLE
updated to 2023-08-01-preview swagger changes . perimeter access rule…

### PR DESCRIPTION
**Pull request by bhupeshbhatt1985**
Description:
Added support for service tag based perimeter access rule.
Swagger updated to 2023-08-01-preview swagger changes . a

---

This checklist is used to make sure that common guidelines for a pull request are followed.
aaz PR: https://github.com/Azure/aaz/pull/507

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/nsp/HISTORY.rst [hunk 1]
```diff
@@ -4,6 +4,13 @@ Release History
 ===============
 ===============
 
+##### 0.3.1
```
Comment (AllyW) [inline src/nsp/HISTORY.rst:7]: version should be updated in setup.py and follow cli extension version schema here: https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md

**Comment by AllyW** [comment]
> ## ⚠️ Release Suggestions
> ### Module: nsp
> * ⚠️ Please update `VERSION` to be `1.0.0b1` in `src/nsp/setup.py`
> 
> #### Notes
> * Stable/preview tag is inherited from last release. If needed, please add `stable`/`preview` label to modify it.
> * Major/minor/patch/pre increment of version number is calculated by pull request code changes automatically. If needed, please add `major`/`minor`/`patch`/`pre` label to adjust it.
> * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

@bhupeshbhatt1985  please refer to this link for correcting module version update, thanks

**Comment by bhupeshbhatt1985** [comment]
> > ## ⚠️ Release Suggestions
> > ### Module: nsp
> > 
> > * ⚠️ Please update `VERSION` to be `1.0.0b1` in `src/nsp/setup.py`
> > 
> > #### Notes
> > 
> > * Stable/preview tag is inherited from last release. If needed, please add `stable`/`preview` label to modify it.
> > * Major/minor/patch/pre increment of version number is calculated by pull request code changes automatically. If needed, please add `major`/`minor`/`patch`/`pre` label to adjust it.
> > * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)
> 
> @bhupeshbhatt1985 please refer to this link for correcting module version update, thanks

@AllyW  updated to [1.0.0b1] as per suggestion and understanding from the document. Bot has removed the block tag after that but ci failing . please review and suggest if this is correct or not.

**Review by AllyW — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Comment by bhupeshbhatt1985** [comment]
@kairu-ms Please don't release it. I need to make a fix.

**Comment by bhupeshbhatt1985** [comment]
@kairu-ms  I raised another PR , fixing the issue. https://github.com/Azure/azure-cli-extensions/pull/7769 .

**Pull request merged**